samcli/local/init/templates/cookiecutter-aws-sam-hello-powershell/{{cookiecutter.project_name}}/README.md

```diff
@@ -0,0 +1,149 @@
+
+## Setup process
+
+### Linux & macOS
+
+```bash
+sh build.sh --target=Package
+```
+
+### Windows (Powershell)
+
+```powershell
+build.ps1 --target=Package
+```
+
+### Local development
+
+**Invoking function locally through local API Gateway**
+
+```bash
+sam local start-api
+```
+
+**SAM Local** is used to emulate both Lambda and API Gateway locally and uses our `template.yaml` to understand how to bootstrap this environment (runtime, where the source code is, etc.) - The following excerpt is what the CLI will read in order to initialize an API and its routes:
+
+```yaml
+...
+Events:
+    HelloWorldFunction:
+        Type: Api # More info about API Event Source: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#api
+        Properties:
+            Path: /hello
+            Method: get
+```
+
+
+If the previous command run successfully you should now be able to hit the following local endpoint to invoke your function `http://localhost:3000/hello`
+
+## Packaging and deployment
+
+AWS Lambda C# runtime requires a flat folder with all dependencies including the application. SAM will use `CodeUri` property to know where to look up for both application and dependencies:
+
+```yaml
+...
+    HelloWorldFunction:
+        Type: AWS::Serverless::Function
+        Properties:
+            CodeUri: artifacts/HelloWorld.zip            
+            ...
+```
+
+First and foremost, we need an `S3 bucket` where we can upload our Lambda functions packaged as ZIP before we deploy anything - If you don't have a S3 bucket to store code artifacts then this is a good time to create one:
+
+```bash
+aws s3 mb s3://BUCKET_NAME
+```
+
+Next, run the following command to package our Lambda function to S3:
+
+```bash
+sam package \
+    --template-file template.yaml \
+    --output-template-file packaged.yaml \
+    --s3-bucket REPLACE_THIS_WITH_YOUR_S3_BUCKET_NAME
+```
+
+Next, the following command will create a Cloudformation Stack and deploy your SAM resources.
+
+```bash
+sam deploy \
+    --template-file packaged.yaml \
+    --stack-name {{ cookiecutter.project_name.lower().replace(' ', '-') }} \
+    --capabilities CAPABILITY_IAM
+```
+
+> **See [Serverless Application Model (SAM) HOWTO Guide](https://github.com/awslabs/serverless-application-model/blob/master/HOWTO.md) for more details in how to get started.**
+
+After deployment is complete you can run the following command to retrieve the API Gateway Endpoint URL:
+
+```bash
+aws cloudformation describe-stacks \
+    --stack-name {{ cookiecutter.project_name.lower().replace(' ', '-') }} \
+    --query 'Stacks[].Outputs'
+```
+## Testing
+
+For testing our code, we use XUnit and you can use `dotnet test` to run tests defined under `test/`
+
+```bash
+dotnet test test/HelloWorld.Test
+```
+
+Alternatively, you can use Cake. It discovers and executes all the tests.
+
+### Linux & macOS
+
+```bash
+sh build.sh --target=Test
```

Comment:
I haven't tried running Powershell or this example yet, but is there any special configuration to be made other than selecting dotnetcore2.1 as a runtime and running Cake for restoring deps?

I guess I'm trying to think if there are other areas we need to cover if those customers are running powershell in Lambda for the first time

Reply:
I've update the readme in the example which now spells out how to run it - have a look over that and see if that helps answer your concerns